pandas/core/window/rolling.py

```diff
@@ -1171,8 +1171,6 @@ class _Rolling_and_Expanding(_Rolling):
     )
 
     def count(self):
-        if isinstance(self.window, BaseIndexer):
-            validate_baseindexer_support("count")
```

Comment:
Custom `BaseIndexer` implementations should never go into this function. Instead, `_get_window_indexer` gets called, and there is a `validate_baseindexer_support` there (checked that it works by temporarily blacklisting `count` again)

Removing this check as unnecessary.

Reply:
can u add an assertion here

Reply:
Done.